components/builder-depot/src/upstream.rs

```diff
@@ -163,36 +168,40 @@ impl UpstreamMgr {
                     continue;
                 }
 
-                let msg_ident: OriginPackageIdent =
+                let upstream_request: UpstreamRequest =
                     parse_from_bytes(&self.msg).map_err(Error::Protobuf)?;
 
-                debug!("Upstream received message: {:?}", msg_ident);
+                let msg_ident = upstream_request.get_ident();
+                let target = upstream_request.get_target();
 
-                // We only care about the base ident
-                let ident =
```

Comment:
The main thing with moving this to check_request is that we could have many different variants of the same package (eg, if the user was browsing a list of versions of a package in the UI) that will trigger N different calls to the upstream.  Though it might be ok since we are discussing switching over to fetching a specific package instead of just the latest..